qucs/extsimkernels/spicecompat.cpp

```diff
@@ -74,6 +75,41 @@ QString spicecompat::normalize_value(QString Value)
     return s.toUpper();
 }
 
+/*!
+ * \brief spicecompat::convert_notation Replaces common unit suffixes and SI
+ *        prefixes in a string with their corresponding exponent values.
+ *        For example, "1.23u" would be converted to "1.23E-06",
+ *        and "1.23m" would be converted to "1.23E-03".
+ *
+ *        Supported unit suffixes include "p", "n", "u", "m", "k", "Meg", "G", and "T".
+ *
+ * \param Value The input string to convert.
+ * \return The string with converted notation.
+ */
+QString spicecompat::convert_notation(QString Value)
```

Comment:
This implementation may not work properly if called on variable but not a number with a suffix. It will replace all occurence of suffix (for example "P" letter). The proper implementation could be done with regexp.